internal/pkg/dashboard/apidata/source.go

```diff
@@ -85,7 +86,30 @@ func (source *Source) run(dataCh chan<- *Data) {
 	}
 }
 
-//nolint:gocyclo,cyclop
+type protoMsg[T any] interface {
+	GetMessages() []T
+}
+
+func unpack[T helpers.Message](source *Source, nodes map[string]*Node, resultLock *sync.Mutex, resp protoMsg[T], setter func(node *Node, value T)) {
+	resultLock.Lock()
+	defer resultLock.Unlock()
+
+	for _, msg := range resp.GetMessages() {
+		node := source.node(msg)
+
+		if _, ok := nodes[node]; !ok {
+			nodes[node] = &Node{}
+		}
+
+		if msg.GetMetadata().GetError() != "" {
```

Comment:
this is the most critical fix, which is hard to see as I de-duplicated all the code